enterprise/decorators.py

```diff
@@ -134,11 +140,15 @@ def wrapper(request, *args, **kwargs):
         # Enterprise-linked IdP and the pipeline will get them back here.
         if not request.user.is_authenticated():
             parsed_current_url = urlparse(request.get_full_path())
-            parsed_query_string = parse_qs(parsed_current_url.query)
-            parsed_query_string.update({
+            parsed_query_string = parse_qsl(parsed_current_url.query)
+
+            additional_query_parameters = {
                 'tpa_hint': enterprise_customer.identity_provider,
                 FRESH_LOGIN_PARAMETER: 'yes'
-            })
+            }
+            for query_parameter_key, query_parameter_value in iteritems(additional_query_parameters):
+                parsed_query_string.append((query_parameter_key, query_parameter_value))
+
             next_url = '{current_path}?{query_string}'.format(
                 current_path=quote(parsed_current_url.path),
                 query_string=urlencode(parsed_query_string)
```

Comment:
Instead of using `parse_qsl` above, just add `doseq=True` to the call to `urlencode` here.

Reply:
@douglashall Thanks for this nice suggestion: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode